### PR TITLE
[Enhancement][Infrastructure] Scientific Linux support

### PR DESCRIPTION
**Pull request by redhatrises**
- add ssg-sl (scientific linux) variant

### CODE REVIEW

**Comment by shawndwells** [comment]
We'll quickly assume a non-trivial amount of technical debt if we maintain scripts for each derivative.

Could the existing enable-centos.py script be reworked as "enable-derivatives.py"?

**Comment by redhatrises** [comment]
> Could the existing enable-centos.py script be reworked as "enable-derivatives.py"? 

I think it should. I can re-work it. Can the re-work be in a separate PR or do you want it in this one?

File: shared/utils/enable-sl.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    "that produce false positives and/or false negatives. If this occurs " \
+    "please file a bug report.</li>\n" \
+    "\n" \
+    "<li><i>Scientifc Linux</i> has its own build system, compiler options, patchsets, " \
+    "and is a community supported, non-commercial operating system. " \
+    "<i>Scientifc Linux</i> does not inherit " \
+    "certifications or evaluations from <i>Red Hat Enterprise Linux</i>. As " \
+    "such, some configuration rules (such as those requiring " \
+    "<i>FIPS 140-2</i> encryption) will continue to fail on <i>Scientifc Linux</i>.</li>\n" \
+    "</ul>\n" \
```
Comment (shawndwells) [inline shared/utils/enable-sl.py]: We can update the caveats using language pulled from the Scientific Linux wikipedia:

`````
- Scientific Linux is not an exact copy of Red Hat Enterprise Linux. Scientific Linux is a Linux distribution produced by
Fermi National Accelerator Laboratory. It is a free and open source operating system based on Red Hat Enterprise Linux
and aims to be "as close to the commercial enterprise distribution as we can get it." There may be configuration differences
that produce false positives and/or false negatives. If this occurs, please file a bug report.

- Scientific Linux is derived from the free and open source software made available by Red Hat, but it is not produced, maintained or supported by Red Hat. Scientific Linux has its own build systems, compiler options, patchsets, packages, and is a community supported, non-commercial operating system. Scientific Linux does not inherit certifications or evaluations from Red Hat Enterprise Linux. As such, some configuration rules (such as those requiring FIPS 140-2 encryption) will continue to fail on Scientific Linux.
`````

ref http://en.wikipedia.org/wiki/Scientific_Linux

**Comment by shawndwells** [comment]
On 5/28/15 1:16 PM, redhatrises wrote:
>
>     Could the existing enable-centos.py script be reworked as
>     "enable-derivatives.py"?
>
> I think it should. I can re-work it. Can the re-work be in a separate 
> PR or do you want it in this one?
>

Doesn't matter.

**Comment by redhatrises** [comment]
> We can update the caveats using language pulled from the Scientific Linux wikipedia

@shawndwells have updated per your request.

**Comment by redhatrises** [comment]
>>
>>     Could the existing enable-centos.py script be reworked as
>>     "enable-derivatives.py"?
>>
>> I think it should. I can re-work it. Can the re-work be in a separate 
>> PR or do you want it in this one?
>>

> Doesn't matter.

@shawndwells I will do the re-work in a separate PR then.

**Comment by mpreisler** [comment]
I recommend first getting SL support into openscap inbuilt CPE. After that's done we can just rename and slightly refactor the enable-centos.py script. Prevent a lot of code duplication and maintenance pain. Do you think it's better to add the CentOS and SL CPEs to one set of files or should we duplicate the files? We can probably make the notice generic enough so that we only need ssg-derivatives6-ds.xml etc...

The folder with inbuilt CPE dict and OVAL: https://github.com/OpenSCAP/openscap/tree/maint-1.2/cpe

I recommend a pull request against branch 1.2.

**Comment by redhatrises** [comment]
> I recommend a pull request against branch 1.2.

@mpreisler PR created: https://github.com/OpenSCAP/openscap/pull/83

> Do you think it's better to add the CentOS and SL CPEs to one set of files or should we duplicate the files?

The `enable-derivatives.py` to be created after this is merged should contain both CentOS and SL contents which would be the one set of files. After that, I would probably say that they should be separated files since the "derivatives" of Red Hat Enterprise Linux may not be equal implementations of each other.

**Comment by mpreisler** [comment]
One other solution is to refactor the functionality of `enable-centos.py` into a package called `derivatives_support`. Then both `enable-centos.py` and `enable-sl.py` can import this package and reuse the same code. Or rename `enable-centos.py` to `enable-derivatives.py` and add a 3rd argument which specifies which derivatives we are building content for.

Either way, I would prefer not to merge a pull request that duplicates code.

**Comment by redhatrises** [comment]
Thanks @mpreisler. I added the `enable-derivatives.py` script and updated the Makefiles.

**Comment by mpreisler** [comment]
ACK.

I think using *optparse* is unnecessary but at the same time it doesn't hurt anything so I won't block the merge with that.

Tested the script on RHEL6 and it works as expected.

**Pull request merged**